utils/markdown/renderer.tsx

```diff
@@ -188,7 +186,7 @@ export default class Renderer extends marked.Renderer {
     );
     internalLink = pattern.test(outHref);
 
-    if (internalLink) {
+    if (internalLink && this.formattingOptions.siteURL) {
```

Comment:
I've added `this.formattingOptions.siteURL`, because TS throw an error `.siteURL` might be `null`.